hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -1076,11 +1076,12 @@ namespace hazelcast {
                     // for back pressure
                     call_id_sequence_->next();
                     invoke_on_selection();
-                    return invocation_promise_.get_future().then(boost::launch::sync,
-                                                               [=](boost::future<protocol::ClientMessage> f) {
-                                                                   call_id_sequence_->complete();
-                                                                   return f.get();
-                                                               });
+                    auto id_seq = call_id_sequence_;
+                    return invocation_promise_.get_future().then(boost::launch::async,
```

Comment:
If I'm not mistaken, a new thread will be created for every invocation. Right?

Reply:
yes, that is true for most implementations.